.github/scripts/generate-matrix.sh

```diff
@@ -0,0 +1,10 @@
+event_name="$1"
+
+if [[ "$event_name" == "push" ]] || [[ "$event_name" == "pull_request" ]]; then
```

Comment:
For a push to master I would run all tests.

Reply:
Agree. That would mainly automate it.

- only run a few select tests for PR commits.
- run all tests when merging into master.

Reply:
Okay, I have updated the matrix script to first check "is it a manual workflow dispatch OR a request from master branch" and if so fun all tests. In all other cases (push/PR/non master) run the subset of tests.